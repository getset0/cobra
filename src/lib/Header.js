import styled from "styled-components";
import React from "react";

const LogoWrapper = styled.a`
  color: inherit;
  text-decoration: none;
`;

const CustomHeader = ({ Logo, user, ...props }) => (
  <header {...props}>
    <div>
      <LogoWrapper href="/">
        {Logo || "Brand"}
      </LogoWrapper>
    </div>
    <div>
      {user || "John Doe"}
    </div>
  </header>
);

const Header = styled(CustomHeader)`
  background-color: ${props => props.theme.whiteColor};
  border-bottom: 1px solid ${props => props.theme.lightGrayColor};
  color: ${props => props.theme.primaryColor};
  position: relative;
  padding: 10px 15px;
  display: flex;
  height: 40px;
  align-items: center;
  justify-content: space-between
`;

export default Header;
