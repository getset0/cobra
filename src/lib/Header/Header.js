import styled from "styled-components";
import React from "react";
import PropTypes from "prop-types";
import { theme } from "../constants";

const HeaderItemWrapper = styled.a`
  text-decoration: none;
  line-height: 40px;
  color: ${props => props.theme.primaryColor};
`;

const CustomHeader = ({ logo, headerCustomRight, user, theme, ...props }) => (
  <header {...props}>
    <div>
      <HeaderItemWrapper href="/" theme={theme}>
        {logo}
      </HeaderItemWrapper>
    </div>
    {headerCustomRight ? (
      <div>{headerCustomRight}</div>
    ) : user ? (
      <HeaderItemWrapper>{user}</HeaderItemWrapper>
    ) : null}
  </header>
);

const Header = styled(CustomHeader)`
  background-color: ${props => props.backgroundColor || props.theme.whiteColor};
  border-bottom: 1px solid ${props => props.theme.lightGrayColor};
  position: relative;
  padding: 10px 15px;
  display: flex;
  height: 40px;
  z-index: 98;
  justify-content: space-between;
`;

Header.propTypes = {
  backgroundColor: PropTypes.string,
  user: PropTypes.string,
  logo: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.element
  ]),
  headerCustomRight: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.element
  ]),
  theme: PropTypes.object
};

Header.defaultProps = {
  logo: "Brand",
  theme: theme
};

export default Header;
