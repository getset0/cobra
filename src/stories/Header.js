import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { withBackgrounds } from "@storybook/addon-backgrounds";
import { backgroundOptions } from "./util";

import {
  List,
  ListItem,
  Link,
  Header,
  Dropdown,
  DropdownTrigger,
  DropdownBody,
  DropdownButton
} from "../lib";

storiesOf("Header", module)
  .addDecorator(withInfo)
  .addDecorator(withBackgrounds(backgroundOptions))
  .add("Simple", () => <Header />)
  .add("List", () => (
    <Header
      rightElement={
        <List orientation="horizontal">
          <ListItem>
            <Link href="#">Item</Link>
          </ListItem>
          <ListItem>
            <Link href="#">Item</Link>
          </ListItem>
          <ListItem>
            <Link href="#">Item</Link>
          </ListItem>
        </List>
      }
    />
  ))
  .add("Dropdown", () => (
    <Header
      rightElement={
        <Dropdown withBorder>
          {({ isOpen, handleTriggerClick }) => (
            <React.Fragment>
              <DropdownTrigger onClick={handleTriggerClick}>
                <DropdownButton isOpen={isOpen}>Trigger</DropdownButton>
              </DropdownTrigger>
              <DropdownBody isOpen={isOpen}>
                <List>
                  <ListItem onClick={() => console.log("item1")}>
                    item1
                  </ListItem>
                  <ListItem onClick={() => console.log("item2")}>
                    item2
                  </ListItem>
                  <ListItem onClick={() => console.log("item3")}>
                    item3
                  </ListItem>
                  <ListItem onClick={() => console.log("item4")}>
                    item4
                  </ListItem>
                </List>
              </DropdownBody>
            </React.Fragment>
          )}
        </Dropdown>
      }
    />
  ));
