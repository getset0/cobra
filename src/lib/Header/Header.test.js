import React from "react";
import Header from "./Header";
import "jest-styled-components";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

it("renders header without crashing", () => {
  shallow(<Header />);
});

it("renders header correctly", () => {
  const wrapper = shallow(<Header />);
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});

