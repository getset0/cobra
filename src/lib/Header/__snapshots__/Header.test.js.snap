// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header correctly 1`] = `
.c0 {
  background-color: #ffffff;
  border-bottom: 1px solid #D8D8D8;
  position: relative;
  padding: 10px 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  z-index: 98;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<CustomHeader
  className="c0"
  logo="Brand"
  theme={
    Object {
      "blackColor": "#000000",
      "darkBlueColor": "#02469B",
      "darkGrayColor": "#4A4A4A",
      "darkestBlueColor": "#02254D",
      "darkestGrayColor": "#303030",
      "greenColor": "#76B041",
      "lightBlueColor": "#0190D0",
      "lightGrayColor": "#D8D8D8",
      "lightestBlueColor": "#00FEFE",
      "mediumGrayColor": "#979797",
      "primaryColor": "#1676B7",
      "secondaryColor": "#17BEBB",
      "whiteBackgroundColor": "#fdfdfd",
      "whiteColor": "#ffffff",
      "yellowColor": "#FFC914",
    }
  }
  user="John Doe"
/>
`;
